### PR TITLE
Allow networkpolicy changes for custom ingress

### PR DESCRIPTION
**Pull request by NautiluX**
* Users need to be allowed to set networkpolicies for custom ingresses in namespace openshift-ingress
* Refactored the test since a lot of new CRUD tests had to be added to cover edge cases

Jira: [OSD-20308](https://issues.redhat.com//browse/OSD-20308)

### CODE REVIEW

File: pkg/webhooks/networkpolicies/networkpolicies.go [hunk 1]
```diff
@@ -9,6 +9,7 @@ import (
 
 	hookconfig "github.com/openshift/managed-cluster-validating-webhooks/pkg/config"
 	"github.com/openshift/managed-cluster-validating-webhooks/pkg/webhooks/utils"
+	k8sapi "k8s.io/api/networking/v1"
```
Comment (mjlshen) [inline pkg/webhooks/networkpolicies/networkpolicies.go:12]:
```suggestion
	networkingv1 "k8s.io/api/networking/v1"
```
Definitely a nit, but there are a lot of k8s api imports, this would align with the other imports below

Comment (NautiluX) [reply pkg/webhooks/networkpolicies/networkpolicies.go]: oh right, thanks for pointing that out

File: pkg/webhooks/networkpolicies/networkpolicies_test.go [hunk 2]
```diff
@@ -1,40 +1,141 @@
 package networkpolicies
 
 import (
-	"fmt"
+	"encoding/json"
+	"strings"
 	"testing"
 
 	admissionv1 "k8s.io/api/admission/v1"
+	k8sapi "k8s.io/api/networking/v1"
```
Comment (mjlshen) [inline pkg/webhooks/networkpolicies/networkpolicies_test.go:9]:
```suggestion
	networkingv1 "k8s.io/api/networking/v1"
```

**Review by mjlshen — APPROVED** [review]
/lgtm
